doc/SAI-Proposal-Global-PTP-Configuration.md

```diff
@@ -0,0 +1,81 @@
+# Global PTP Configuration in SAI Switch
+
+Title       | Global PTP Configuration in SAI Switch
+------------|----------------------------------------
+Authors     | Open
+Status      | Draft
+Type        | Standards track
+Created     | 2024-03-03
+SAI-Version | 1.15
+
+## Overview
+Introducing a PTP (Precision Time Protocol) attribute to the SAI switch object, allowing for switch-wide PTP mode settings that affect all ports on the switch. 
+
+A switch that does not set this attribute will observe no change in behavior.
+
+## Motivation
+Currently, PTP configuration in SAI is port-based, and many switches only support configuration of PTP on a global basis.
+In addition, many existing switches have a configuration model which allows global configuration of PTP, and then override on a port level.  SAI should follow this flexible example.
+
+## Technical Specification
+
+### New Switch PTP Configuration Type
+```c
+typedef struct _sai_switch_ptp_mode_t
+{
```

Comment:
All the comments that are addressed in the C header files needs to be reflect in this MD file as well.